xdsl/ir/affine/affine_expr.py

```diff
@@ -159,9 +159,42 @@ def eval(self, dims: list[int], symbols: list[int]) -> int:
 
         raise ValueError("Unreachable")
 
+    def _try_fold_constant(
```

Comment:
Before you merge this, why not make the signature of this function 
```suggestion
    def _try_fold_constant(self) -> AffineExpr:
```
Where you return self if you could not merge the constant? Where the implementation folds the constant in the lhs, rhs, and then matches on the kind of the operation to see how they could be combined?

Reply:
Also it seems like swapping around the order of arguments might be confusing, feels like that's a separate thing from constant folding to me.

Reply:
> Also it seems like swapping around the order of arguments might be confusing, feels like that's a separate thing from constant folding to me.

It's canonicalizing the input for simplifications. Seems pretty standard to me?

Reply:
> Before you merge this, why not make the signature of this function
> 
> Where you return self if you could not merge the constant? Where the implementation folds the constant in the lhs, rhs, and then matches on the kind of the operation to see how they could be combined?

I like the current implementation. It allows me to fold things like (expr + constant) + constant too.